PhysicsTools/NanoAOD/plugins/CandMCMatchTableProducer.cc

```diff
@@ -114,8 +114,10 @@ class CandMCMatchTableProducer : public edm::global::EDProducer<> {
             flav[i] = getParentHadronFlag(match);  // 3 = light, 4 = charm, 5 = b
           break;
         case MPhoton:
-          if (match->isPromptFinalState())
-            flav[i] = (match->pdgId() == 22 ? 1 : 13);  // prompt electron or photon
+          if (match->isPromptFinalState() && match->pdgId() == 22)
+            flav[i] = 1;  // prompt photon
+          else if ((match->isPromptFinalState() || match->isDirectPromptTauDecayProductFinalState()) && abs(match->pdgId()) == 11)
+            flav[i] = 13; // prompt electron
```

Comment:
I'd use this opportunity to assign 11 instead of 13 as the flavor ID because 13 usually refers to muons.